lib/fluent/plugin_helper/event_emitter.rb

```diff
@@ -26,10 +26,14 @@ module EventEmitter
 
       def router
         @_event_emitter_used_actually = true
+        if @_event_emitter_lazy_init
+          @router = @primary_instance.router
+        end
         @router
       end
 
       def router=(r)
+        # not recommended now...
```

Comment:
I thought about it at first, but didn't it finally. Because we can't remove `#router=` method between v0.x and v1.x for compatibility reason. So displaying warning doesn't help anything.
IMO we should show warning here at the same time with deprecating v0.12 plugin APIs (and remove this method at v2).